src/items/associated-items.md

```diff
@@ -109,13 +109,15 @@ following types:
 - [`Arc<Self>`]
 - [`Pin<P>`] where `P` is one of the above types except `Self`.
 
-The `Self` term can be replaced with the type being implemented.
+The `Self` term can be replaced with the type being implemented, including
+type aliases for the type, or any nested combination of the above types.
```

Comment:
Having another look at this now, I wonder if we could rephrase this section more formally as (with formatting to be fixed):

> If the type of the self parameter is specified, it is limited to *semantic* types generated by the following grammar (where `'lt` denotes some arbitrary lifetime):
>
> ```
> P = &'lt S | &'lt mut S | Box<S> | Rc<S> | Arc<S> | Pin<P>
> S = Self | P
> ```
>
> The `Self` terminal in this grammar is the semantic `Self` type and can be replaced with the type being implemented, including type aliases or associated type projections for the type.